### PR TITLE
Move theming js to initial state

### PR DESCRIPTION
**Pull request by rullzer**
A bit more elegant. Plus it will allow us to also write a proper
@nextcloud/theming package. To make life easier down the line for all.

Huge plus here is that the file is served always statically and thus bypasses Nextcloud all together.

### CODE REVIEW

File: apps/theming/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -24,8 +24,29 @@
...
+
 	public function __construct() {
-		parent::__construct('theming', []);
+		parent::__construct(self::APP_ID);
+
+		$container = $this->getContainer();
+		$this->regsiterInitialState($container);
+	}
+
+	private function regsiterInitialState(IAppContainer $container) {
```
Comment (tcitworld) [inline apps/theming/lib/AppInfo/Application.php:42]:
```suggestion
	private function registerInitialState(IAppContainer $container) {
```

File: apps/theming/lib/AppInfo/Application.php [hunk 2]
```diff
@@ -24,8 +24,29 @@
...
 class Application extends \OCP\AppFramework\App {
+
+	public const APP_ID = 'theming';
+
 	public function __construct() {
-		parent::__construct('theming', []);
+		parent::__construct(self::APP_ID);
+
+		$container = $this->getContainer();
+		$this->regsiterInitialState($container);
```
Comment (tcitworld) [inline apps/theming/lib/AppInfo/Application.php:39]:
```suggestion
		$this->registerInitialState($container);
```

**Comment by kesselb** [comment]
Another plus is: minus one usage of url generator on app initialization. That's good because the url generator always loads other apps. 

https://github.com/nextcloud/server/blob/master/apps/accessibility/lib/AppInfo/Application.php would be another candidate ;)

File: apps/theming/lib/Service/JSDataService.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		Util $util,
+		IConfig $appConfig
+	) {
+
+		$this->themingDefaults = $themingDefaults;
+		$this->util = $util;
+		$this->appConfig = $appConfig;
+	}
+
+	public function getData(): array  {
```
Comment (ChristophWurst) [inline apps/theming/lib/Service/JSDataService.php:53]: bonus points if you use JsonSerializable instead :)

**Review by ChristophWurst — APPROVED** [review]

**Comment by rullzer** [comment]
> Another plus is: minus one usage of url generator on app initialization. That's good because the url generator always loads other apps.

yeah... well I mean we do the same for the css... So there is that..

> https://github.com/nextcloud/server/blob/master/apps/accessibility/lib/AppInfo/Application.php would be another candidate ;)

yep also on the list

File: apps/theming/lib/Service/JSDataService.php (hunk 3 above)
Comment (rullzer) [reply apps/theming/lib/Service/JSDataService.php]: yes of course. done

**Review by georgehrke — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Comment by ChristophWurst** [comment]
> Lint / php-cs check

fixed

**Pull request merged**